llap-tez/src/java/org/apache/hadoop/hive/llap/tezplugins/LlapTaskSchedulerService.java

```diff
@@ -1175,6 +1178,9 @@ public boolean deallocateTask(
       LOG.debug("Processing deallocateTask for task={}, taskSucceeded={}, endReason={}", task,
           taskSucceeded, endReason);
     }
+    if (task instanceof TaskAttemptImpl && metrics != null) {
+      updateMetrics((TaskAttemptImpl)task);
```

Comment:
The metrics can be null in case of conf.getBoolean(ConfVars.HIVE_IN_TEST.varname, false)

I tried to find the style problem, but not sure which part should be written differently